test/e2e/scale/scale_test.go

```diff
@@ -0,0 +1,202 @@
+	"sigs.k8s.io/azuredisk-csi-driver/test/utils/credentials"
+)
+
+var (
+	skipClusterBootstrap = flag.Bool("skip-cluster-bootstrap", true, "flag to indicate that we can skip cluster bootstrap.")
+)
+
+var _ = ginkgo.BeforeSuite(func() {
+	// k8s.io/kubernetes/test/e2e/framework requires env KUBECONFIG to be set
+	// it does not fall back to defaults
+	if os.Getenv(e2eutils.KubeconfigEnvVar) == "" {
+		kubeconfig := filepath.Join(os.Getenv("HOME"), ".kube", "config")
+		os.Setenv(e2eutils.KubeconfigEnvVar, kubeconfig)
+	}
+	handleFlags()
+	framework.AfterReadingAllFlags(&framework.TestContext)
+
+	// Default storage driver configuration is CSI. Freshly built
+	// CSI driver is installed for that case.
+	if e2eutils.IsTestingMigration || !e2eutils.IsUsingInTreeVolumePlugin {
+		creds, err := credentials.CreateAzureCredentialFile()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		azureClient, err := azure.GetAzureClient(creds.Cloud, creds.SubscriptionID, creds.AADClientID, creds.TenantID, creds.AADClientSecret)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		_, err = azureClient.EnsureResourceGroup(context.Background(), creds.ResourceGroup, creds.Location, nil)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Install Azure Disk CSI Driver on cluster from project root
+		e2eBootstrap := e2eutils.TestCmd{
+			Command:  "make",
+			Args:     []string{"e2e-bootstrap"},
+			StartLog: "Installing Azure Disk CSI Driver...",
+			EndLog:   "Azure Disk CSI Driver installed",
+		}
+
+		createMetricsSVC := e2eutils.TestCmd{
+			Command:  "make",
+			Args:     []string{"create-metrics-svc"},
+			StartLog: "create metrics service ...",
+			EndLog:   "metrics service created",
+		}
+		if !*skipClusterBootstrap {
+			e2eutils.ExecTestCmd([]e2eutils.TestCmd{e2eBootstrap, createMetricsSVC})
+		}
+	}
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	// Default storage driver configuration is CSI. Freshly built
+	// CSI driver is installed for that case.
+	if e2eutils.IsTestingMigration || e2eutils.IsUsingInTreeVolumePlugin {
+		cmLog := e2eutils.TestCmd{
+			Command:  "bash",
+			Args:     []string{"test/utils/controller-manager-log.sh"},
+			StartLog: "===================controller-manager log=======",
+			EndLog:   "===================================================",
+		}
+		e2eutils.ExecTestCmd([]e2eutils.TestCmd{cmLog})
+	}
+
+	if e2eutils.IsTestingMigration || !e2eutils.IsUsingInTreeVolumePlugin {
+		checkPodsRestart := e2eutils.TestCmd{
+			Command:  "bash",
+			Args:     []string{"test/utils/check_driver_pods_restart.sh", "log"},
+			StartLog: "Check driver pods if restarts ...",
+			EndLog:   "Check successfully",
+		}
+		e2eutils.ExecTestCmd([]e2eutils.TestCmd{checkPodsRestart})
+
+		os := "linux"
+		cloud := "azurepubliccloud"
+		if e2eutils.IsWindowsCluster {
+			os = "windows"
+		}
+		if e2eutils.IsAzureStackCloud {
+			cloud = "azurestackcloud"
+		}
+		createExampleDeployment := e2eutils.TestCmd{
```

Comment:
I don't think we need to create all these resources as a final verification in the scale test. These are already checked as a part of e2e.